src/main/java/com/limechain/runtime/hostapi/MiscellaneousHostFunctions.java

```diff
@@ -1,67 +1,151 @@
+
+    public MiscellaneousHostFunctions() {
+        this.hostApi = HostApi.getInstance();
+    }
+
     public static List<ImportObject> getFunctions() {
+        return new MiscellaneousHostFunctions().buildFunctions();
+    }
+
+    public List<ImportObject> buildFunctions() {
         return Arrays.asList(
-                HostApi.getImportObject("ext_misc_print_num_version_1", argv -> {
-
-                }, List.of(Type.I64)),
-                HostApi.getImportObject("ext_misc_print_utf8_version_1", argv -> {
-
-                }, List.of(Type.I64)),
-                HostApi.getImportObject("ext_misc_print_hex_version_1", argv -> {
-                    extMiscPrintHex((long) argv.get(0));
-                }, List.of(Type.I64)),
-                HostApi.getImportObject("ext_misc_runtime_version_version_1", argv -> {
-                    return argv;
-                }, List.of(Type.I64), Type.I64),
-                HostApi.getImportObject("ext_logging_log_version_1", argv -> {
-                    extLoggingLog((Integer) argv.get(0), (Long) argv.get(1), (Long) argv.get(2));
-                }, Arrays.asList(Type.I32, Type.I64, Type.I64)),
-                HostApi.getImportObject("ext_logging_max_level_version_1", argv -> {
-                    return argv;
-                }, List.of(), Type.I32),
-                HostApi.getImportObject("ext_panic_handler_abort_on_panic_version_1", argv -> {
-                    extPanicHandlerAbortOnPanicVersion1((Long) argv.get(0));
-                }, List.of(Type.I64)));
+                HostApi.getImportObject("ext_misc_print_num_version_1", argv ->
+                        printNumV1(argv.get(0)), List.of(Type.I64)),
+                HostApi.getImportObject("ext_misc_print_utf8_version_1", argv ->
+                        printUtf8V1(new RuntimePointerSize(argv.get(0))), List.of(Type.I64)),
+                HostApi.getImportObject("ext_misc_print_hex_version_1", argv ->
+                        printHexV1(new RuntimePointerSize(argv.get(0))), List.of(Type.I64)),
+                HostApi.getImportObject("ext_misc_runtime_version_version_1", argv ->
+                                List.of(runtimeVersionV1(new RuntimePointerSize(argv.get(0))).pointerSize()),
+                        List.of(Type.I64), Type.I64),
+                HostApi.getImportObject("ext_logging_log_version_1", argv ->
+                        logV1(argv.get(0).intValue(), new RuntimePointerSize(argv.get(1)),
+                                new RuntimePointerSize(argv.get(1))), Arrays.asList(Type.I32, Type.I64, Type.I64)),
+                HostApi.getImportObject("ext_logging_max_level_version_1", argv ->
+                        List.of(maxLevelV1()), List.of(), Type.I32)
+        );
+    }
+
+    /**
+     * Print a number.
+     *
+     * @param number the number to be printed
+     */
+    public void printNumV1(Number number) {
+        log.info("Printing number from runtime: ");
+        System.out.println(number);
```

Comment:
Why not have this all in the logging line? Do we need to call System here?

Reply:
Not necessarily required to use System output, can add it to the logger's info log.